### PR TITLE
`component` with builtin `fgo`

### PR DESCRIPTION
**Pull request by limemloh**
`component` also takes generator functions and pass them to `fgo` #4

### CODE REVIEW

**Comment by paldepind** [comment]
Great!

Now we just need for the components themselves  to accept a generator function as children in order to close #4.

**Comment by limemloh** [comment]
the `e()` now supports a generator function as a child

File: src/component.ts [hunk 1]
```diff
@@ -95,14 +96,21 @@ class MfixNow<M extends BehaviorObject, O> extends Now<[M, O]> {
   };
 }
 
+export function isGeneratorFunction<A, T>(fn: any): fn is ((a: A) => Iterator<T>) {
+  return fn !== undefined
+    && fn.constructor !== undefined
+    && fn.constructor.name === "GeneratorFunction";
```
Comment (paldepind) [inline src/component.ts]: Cool! I had no idea that you could identify a generator function like that.

**Comment by paldepind** [comment]
Really sweet :great: :tada:

**Pull request merged**